src/wallet.rs

```diff
@@ -70,6 +161,150 @@ impl WalletVersion {
     }
 }
 
+/// WalletVersion::V1R1 | WalletVersion::V1R2 | WalletVersion::V1R3 | WalletVersion::V2R1 | WalletVersion::V2R2
+pub struct DataV1R1 {
+    pub seqno: u32,
+    pub public_key: [u8; 32],
+}
+
+impl TryFrom<&Cell> for DataV1R1 {
+    type Error = TonCellError;
+
+    fn try_from(value: &Cell) -> Result<Self, Self::Error> {
+        let mut parser = value.parser();
+        let seqno = parser.load_u32(32)?;
+        let mut public_key = [0u8; 32];
+        parser.load_slice(&mut public_key)?;
+        Ok(Self { seqno, public_key })
+    }
+}
+
+impl TryInto<Cell> for DataV1R1 {
```

Comment:
I would suggest to follow guideline in rust [documentation](https://doc.rust-lang.org/std/convert/trait.TryInto.html) ,  which recommends to use ` impl From` and `impl TryFrom` instead of  `impl Into` and `impl TryInto`respectively.